h2/src/main/org/h2/mvstore/FreeSpaceBitSet.java

```diff
@@ -114,20 +127,29 @@ public long allocate(int length) {
      * @return the starting block index
      */
     long predictAllocation(int blocks) {
-        return allocate(blocks, false);
+        return allocate(blocks, 0, 0, false);
     }
 
     boolean isFragmented() {
         return Integer.bitCount(failureFlags & 0x0F) > 1;
     }
 
-    private int allocate(int blocks, boolean allocate) {
+    private int allocate(int blocks, int reservedLow, int reservedHigh, boolean allocate) {
         int freeBlocksTotal = 0;
         for (int i = 0;;) {
             int start = set.nextClearBit(i);
```

Comment:
What 's the max value of i? In a test, if i is Integer.MAX_VALUE, and this bit set, then return Integer.MIN_VALUE.

Reply:
Of course, it's a known limitation at this point, but please calculate required db size (16TB). I agree that It's better to put an explicit upper bound here, and give a meaningful error message, instead if facing database corruption.

Reply:
Actually, if on allocation attempt block index would wrap around in
https://github.com/h2database/h2database/blob/2c31396bc9ae68b3648b1e1f11e699aeab57de7b/h2/src/main/org/h2/mvstore/FreeSpaceBitSet.java#L134 (in start + blocks), IndexOutOfBoundsException will be thrown by BitSet, and database would panic, so corruption should not occur. Given **extremely** remote possibility for this to happen, lack of clear error message might be ok, I guess.